tap-agent/src/tap/sender_account.rs

```diff
@@ -0,0 +1,969 @@
+                .allocations
+                .lock()
+                .unwrap()
+                .values()
+                .filter(|a| matches!(a, AllocationState::Ineligible(_)))
+                .map(|a| a.as_ineligible().unwrap())
+                .cloned()
+                .collect::<Vec<_>>();
+
+            for allocation in allocations_finalizing {
+                if let Err(e) = allocation.rav_requester_single().await {
+                    error!(
+                        "Error while requesting RAV for sender {} and allocation {}: {}",
+                        self.sender,
+                        allocation.get_allocation_id(),
+                        e
+                    );
+                    continue;
+                }
+
+                if let Err(e) = allocation.mark_rav_final().await {
+                    error!(
+                        "Error while marking allocation {} as final for sender {}: {}",
+                        allocation.get_allocation_id(),
+                        self.sender,
+                        e
+                    );
+                    continue;
+                }
+
+                // Remove the allocation from the finalizing map.
+                self.allocations
+                    .lock()
+                    .unwrap()
+                    .remove(&allocation.get_allocation_id());
+            }
+        }
+    }
+
+    /// Does a single RAV request for the sender's allocation with the highest unaggregated fees
+    async fn rav_requester_single(&self) {
+        let heaviest_allocation = match self.get_heaviest_allocation() {
+            Ok(a) => a,
+            Err(e) => {
+                error!(
+                    "Error while getting allocation with most unaggregated fees: {}",
+                    e
+                );
+                return;
+            }
+        };
+
+        if let Err(e) = heaviest_allocation.rav_requester_single().await {
+            error!(
+                "Error while requesting RAV for sender {} and allocation {}: {}",
+                self.sender,
+                heaviest_allocation.get_allocation_id(),
+                e
+            );
+            return;
+        };
+
+        if let Err(e) = self.recompute_unaggregated_fees().await {
+            error!(
+                "Error while recomputing unaggregated fees for sender {}: {}",
+                self.sender, e
+            );
+        }
+    }
+
+    /// Returns the allocation with the highest unaggregated fees value.
+    fn get_heaviest_allocation(&self) -> Result<Arc<SenderAllocation>> {
+        // Get a quick snapshot of all allocations. They are Arcs, so it should be cheap,
+        // and we don't want to hold the lock for too long.
+        let allocations: Vec<_> = self.allocations.lock().unwrap().values().cloned().collect();
+
+        // Get the fees for each allocation in parallel. This is required because the
+        // SenderAllocation's fees is behind a Mutex.
```

Comment:
We can remove this comment

Reply:
Indeed!

Reply:
Fix in 9c35d85203b854e292fd91c464cd4f1acbf15a2e